### ISSUE TITLE
allow to have missing values on init

### ISSUE THREAD

**Issue by YoniChechik** [pydantic/pydantic#9775]
### Initial Checks

- [X] I have searched Google & GitHub for similar requests and couldn't find anything
- [X] I have read and followed [the docs](https://docs.pydantic.dev) and still think this feature is missing

### Description

current code cause an error
```python
from pydantic import BaseModel


class Scheme(BaseModel):
    x: str
    y: int = None


if __name__ == "__main__":
    config = Scheme()
```
```bash
1 validation error for Scheme
x
  Field required [type=missing, input_value={}, input_type=dict]
    For further information visit https://errors.pydantic.dev/2.7/v/missing
```
I want to have a way to allow for missing values on init and validate them later (on freeze for example)

### Affected Components

- [ ] [Compatibility between releases](https://docs.pydantic.dev/changelog/)
- [X] [Data validation/parsing](https://docs.pydantic.dev/concepts/models/#basic-model-usage)
- [ ] [Data serialization](https://docs.pydantic.dev/concepts/serialization/) - `.model_dump()` and `.model_dump_json()`
- [ ] [JSON Schema](https://docs.pydantic.dev/concepts/json_schema/)
- [X] [Dataclasses](https://docs.pydantic.dev/concepts/dataclasses/)
- [X] [Model Config](https://docs.pydantic.dev/concepts/config/)
- [X] [Field Types](https://docs.pydantic.dev/api/types/) - adding or changing a particular data type
- [X] [Function validation decorator](https://docs.pydantic.dev/concepts/validation_decorator/)
- [ ] [Generic Models](https://docs.pydantic.dev/concepts/models/#generic-models)
- [ ] [Other Model behaviour](https://docs.pydantic.dev/concepts/models/) - `model_construct()`, pickling, private attributes, ORM mode
- [ ] [Plugins](https://docs.pydantic.dev/) and integration with other tools - mypy, FastAPI, python-devtools, Hypothesis, VS Code, PyCharm, etc.

**Comment by sydney-runkle** [comment]
Does this work for your use case?

```py
from pydantic import BaseModel, SkipValidation


class Scheme(BaseModel):
    x: str | None = None
    y: int | None = None


print(repr(Scheme()))
#> Scheme(x=None, y=None)
```

Then you can perform other validations on `x` down the line whenever you please!

**Comment by YoniChechik** [comment]
I'm aware of this possibility but in my case None can be a valid response for some values but not for other.

**Comment by sydney-runkle** [comment]
This reminds me a bit of https://github.com/pydantic/pydantic/issues/8394

**Comment by sydney-runkle** [comment]
Could you use some sentinel value like `_Unset` as the default?

**Comment by YoniChechik** [comment]
I'd like to do so, is this possible? If i declare something like `Unset=None`  i'd again need to change all of my typings. 
My reason behind this is to build something like omegaconf with pydantic (pydentic baseSetting is not strong enough

**Comment by sydney-runkle** [comment]
Here's one option:

```py
from typing import Any, final

from pydantic import BaseModel

@final
class _UnsetType:
    ...

_Unset: Any = _UnsetType()

class Scheme(BaseModel):
    x: str | None = _Unset
    y: int | None = None


s = Scheme()
print(repr(s))
```

@adriangb might have a good suggestion for an even more type safe approach here.

**Comment by adriangb** [comment]
Yes I think you'd want to go with something like that. You can combine with serialization customization to e.g. make it so `UnsetType` gets skipped during serialization.

**Comment by sydney-runkle** [comment]
Going to close this as not planned for now - I believe the sentinel approach with `_Unset` or something similar is easy enough to implement in user code, and our custom serializers / validators enable a lot of flexibility in this area.
